pkg/cmd/controller/route-controller/route.go

```diff
@@ -0,0 +1,83 @@
+package route
+
+import (
+	"context"
+	"os"
+
+	"github.com/openshift/openshift-controller-manager/pkg/route/ingress"
+	"github.com/openshift/openshift-controller-manager/pkg/route/ingressip"
+	v1 "k8s.io/api/core/v1"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+)
+
+func RunRouteControllerManager(ctx *ControllerContext) (bool, error) {
+	kubeClient, err := ctx.ClientBuilder.Client(infraIngressToRouteControllerServiceAccountName)
+	if err != nil {
+		return true, err
+	}
+	config := ctx.OpenshiftControllerConfig
+	var ingressToRouteController *ingress.Controller
+	var ingressController *ingressip.IngressIPController
+	if ctx.IsControllerEnabled("openshift.io/ingress-ip") {
+		ingressController, err = getIngressIPController(ctx)
+		if err != nil {
+			return false, err
+		}
+	}
+	if ctx.IsControllerEnabled("openshift.io/ingress-to-route") {
+		ingressToRouteController, err = getIngressToRouteController(ctx)
+		if err != nil {
+			return false, err
+		}
+	}
+	routeControllerManager := func(cntx context.Context) {
```

Comment:
this logic should be like originControllerManager below and initialize based on the new route controller init function map

Reply:
startControllers go ahead and start the individual controller, don't we want to start both the controllers in controller-manager code?